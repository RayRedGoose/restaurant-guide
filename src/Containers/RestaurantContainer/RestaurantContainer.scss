.restaurants-container {
  width: 75rem;
  margin: 2rem auto;
}
