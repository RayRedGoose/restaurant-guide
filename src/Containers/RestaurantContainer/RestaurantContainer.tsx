import './RestaurantContainer.scss';
import React, { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import { applyFilters, checkEmptyFilters } from '_utils';
import { IAppStore, IRestaurantObject, IFilters } from 'assets/ts/interfaces';
import RestaurantCard from 'Containers/RestaurantCard/RestaurantCard';
import Pagination from 'Containers/Pagination/Pagination';

const RestaurantContainer: React.FC = () => {
  const {
    restaurants,
    currentPage,
    stateFilter,
    genreFilter,
    attireFilter,
    searchFilter,
  } = useSelector((store: IAppStore) => store);

  const filters: IFilters = {
    stateFilter,
    genreFilter,
    attireFilter,
    searchFilter,
  };

  const [displayed, setDisplayed] = useState<IRestaurantObject[]>([]);

  const shortDisplayed = (): void => {
    const increment: number = 10 * currentPage;
    const shorted: IRestaurantObject[] = restaurants
      .slice(0 + increment, 9 + increment);
    setDisplayed(shorted);
  };

  useEffect(shortDisplayed, [currentPage]);

  const filterRestaurants = (): IRestaurantObject[] => applyFilters(restaurants, filters);
  const checkResult = checkEmptyFilters(filters);

  const usedRestaurants: IRestaurantObject[] = (checkResult)
    ? filterRestaurants()
    : displayed;

  const restaurantCards: JSX.Element[] = usedRestaurants
    .map((rest: IRestaurantObject, ind: number) => (
      <RestaurantCard key={`restaurant${ind}`} restaurant={rest} />
    ));

  return (
    <main className="restaurants-container">
      { restaurantCards }
      { !checkResult && <Pagination /> }
    </main>
  );
};

export default RestaurantContainer;
