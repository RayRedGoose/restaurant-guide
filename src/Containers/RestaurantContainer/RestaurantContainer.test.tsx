/**
* @jest-environment jsdom
*/

import React from 'react';
import mockStore from 'assets/ts/test/mockStore';
import RestaurantContainer from './RestaurantContainer';
import { render, act, RenderResult } from '@testing-library/react';

const mockDispatch = jest.fn();
let mockCurrentPage: number = 0;
let mockDisplayPage: number = mockCurrentPage + 1;
let mockSearchFilter: string = '';

jest.mock('react-redux', () => ({
  useSelector: () => ({
    ...mockStore,
    currentPage: mockCurrentPage,
    displayPage: mockDisplayPage,
    searchFilter: mockSearchFilter
  }),
  useDispatch: () => mockDispatch
}));

describe("RestaurantContainer", () => {
  let wrapper: RenderResult;

  it('should match the snapshot', async () => {
    await act(async () => {
      wrapper = render(<RestaurantContainer />);
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with the next pack of restaurants', async () => {
    mockCurrentPage = 1;
    mockDisplayPage = 2;
    await act(async () => {
      wrapper = render(<RestaurantContainer />);
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if there is filled filter', async () => {
    mockSearchFilter = 'seafood';
    await act(async () => {
      wrapper = render(<RestaurantContainer />);
    });
    expect(wrapper).toMatchSnapshot();
  });
});
