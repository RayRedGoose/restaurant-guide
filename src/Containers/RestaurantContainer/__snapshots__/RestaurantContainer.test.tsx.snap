// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantContainer should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="restaurants-container"
      >
        <section
          class="restaurant-card"
        >
          <header>
            <p
              class="genre"
            >
              Steak
            </p>
            <p
              class="genre"
            >
              American
            </p>
            <p
              class="genre"
            >
              Contemporary
            </p>
            <p
              class="genre"
            >
              Seafood
            </p>
            <p
              class="genre"
            >
              Cafe
            </p>
            <p
              id="attire"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-briefcase fa-w-16 "
                data-icon="briefcase"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M320 336c0 8.84-7.16 16-16 16h-96c-8.84 0-16-7.16-16-16v-48H0v144c0 25.6 22.4 48 48 48h416c25.6 0 48-22.4 48-48V288H320v48zm144-208h-80V80c0-25.6-22.4-48-48-48H176c-25.6 0-48 22.4-48 48v48H48c-25.6 0-48 22.4-48 48v80h512v-80c0-25.6-22.4-48-48-48zm-144 0H192V96h128v32z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="label"
              >
                business casual
              </span>
            </p>
          </header>
          <section
            class="card-description"
          >
            <h2>
              Old Hickory Steakhouse
            </h2>
          </section>
          <section
            class="card-information"
          >
            <p
              class="location"
            >
              Oxon Hill, MD
            </p>
            <p
              class="phone"
            >
              (301) 965-4000
            </p>
          </section>
        </section>
        <footer
          class="pagination"
        >
          <p
            class="pages"
          >
            1 / 3
          </p>
          <button
            id="next-page-btn"
          >
             next page 
          </button>
        </footer>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="restaurants-container"
    >
      <section
        class="restaurant-card"
      >
        <header>
          <p
            class="genre"
          >
            Steak
          </p>
          <p
            class="genre"
          >
            American
          </p>
          <p
            class="genre"
          >
            Contemporary
          </p>
          <p
            class="genre"
          >
            Seafood
          </p>
          <p
            class="genre"
          >
            Cafe
          </p>
          <p
            id="attire"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-briefcase fa-w-16 "
              data-icon="briefcase"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M320 336c0 8.84-7.16 16-16 16h-96c-8.84 0-16-7.16-16-16v-48H0v144c0 25.6 22.4 48 48 48h416c25.6 0 48-22.4 48-48V288H320v48zm144-208h-80V80c0-25.6-22.4-48-48-48H176c-25.6 0-48 22.4-48 48v48H48c-25.6 0-48 22.4-48 48v80h512v-80c0-25.6-22.4-48-48-48zm-144 0H192V96h128v32z"
                fill="currentColor"
              />
            </svg>
            <span
              class="label"
            >
              business casual
            </span>
          </p>
        </header>
        <section
          class="card-description"
        >
          <h2>
            Old Hickory Steakhouse
          </h2>
        </section>
        <section
          class="card-information"
        >
          <p
            class="location"
          >
            Oxon Hill, MD
          </p>
          <p
            class="phone"
          >
            (301) 965-4000
          </p>
        </section>
      </section>
      <footer
        class="pagination"
      >
        <p
          class="pages"
        >
          1 / 3
        </p>
        <button
          id="next-page-btn"
        >
           next page 
        </button>
      </footer>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`RestaurantContainer should match the snapshot if there is filled filter 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="restaurants-container"
      >
        <section
          class="restaurant-card"
        >
          <header>
            <p
              class="genre"
            >
              Steak
            </p>
            <p
              class="genre"
            >
              American
            </p>
            <p
              class="genre"
            >
              Contemporary
            </p>
            <p
              class="genre"
            >
              Seafood
            </p>
            <p
              class="genre"
            >
              Cafe
            </p>
            <p
              id="attire"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-briefcase fa-w-16 "
                data-icon="briefcase"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M320 336c0 8.84-7.16 16-16 16h-96c-8.84 0-16-7.16-16-16v-48H0v144c0 25.6 22.4 48 48 48h416c25.6 0 48-22.4 48-48V288H320v48zm144-208h-80V80c0-25.6-22.4-48-48-48H176c-25.6 0-48 22.4-48 48v48H48c-25.6 0-48 22.4-48 48v80h512v-80c0-25.6-22.4-48-48-48zm-144 0H192V96h128v32z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="label"
              >
                business casual
              </span>
            </p>
          </header>
          <section
            class="card-description"
          >
            <h2>
              Old Hickory Steakhouse
            </h2>
          </section>
          <section
            class="card-information"
          >
            <p
              class="location"
            >
              Oxon Hill, MD
            </p>
            <p
              class="phone"
            >
              (301) 965-4000
            </p>
          </section>
        </section>
        <section
          class="restaurant-card"
        >
          <header>
            <p
              class="genre"
            >
              Japanese
            </p>
            <p
              class="genre"
            >
              Sushi
            </p>
            <p
              class="genre"
            >
              Asian
            </p>
            <p
              class="genre"
            >
              Contemporary
            </p>
            <p
              class="genre"
            >
              Seafood
            </p>
            <p
              id="attire"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-briefcase fa-w-16 "
                data-icon="briefcase"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M320 336c0 8.84-7.16 16-16 16h-96c-8.84 0-16-7.16-16-16v-48H0v144c0 25.6 22.4 48 48 48h416c25.6 0 48-22.4 48-48V288H320v48zm144-208h-80V80c0-25.6-22.4-48-48-48H176c-25.6 0-48 22.4-48 48v48H48c-25.6 0-48 22.4-48 48v80h512v-80c0-25.6-22.4-48-48-48zm-144 0H192V96h128v32z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="label"
              >
                business casual
              </span>
            </p>
          </header>
          <section
            class="card-description"
          >
            <h2>
              Matsuhisa
            </h2>
          </section>
          <section
            class="card-information"
          >
            <p
              class="location"
            >
              Aspen, CO
            </p>
            <p
              class="phone"
            >
              (970) 544-6628
            </p>
          </section>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="restaurants-container"
    >
      <section
        class="restaurant-card"
      >
        <header>
          <p
            class="genre"
          >
            Steak
          </p>
          <p
            class="genre"
          >
            American
          </p>
          <p
            class="genre"
          >
            Contemporary
          </p>
          <p
            class="genre"
          >
            Seafood
          </p>
          <p
            class="genre"
          >
            Cafe
          </p>
          <p
            id="attire"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-briefcase fa-w-16 "
              data-icon="briefcase"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M320 336c0 8.84-7.16 16-16 16h-96c-8.84 0-16-7.16-16-16v-48H0v144c0 25.6 22.4 48 48 48h416c25.6 0 48-22.4 48-48V288H320v48zm144-208h-80V80c0-25.6-22.4-48-48-48H176c-25.6 0-48 22.4-48 48v48H48c-25.6 0-48 22.4-48 48v80h512v-80c0-25.6-22.4-48-48-48zm-144 0H192V96h128v32z"
                fill="currentColor"
              />
            </svg>
            <span
              class="label"
            >
              business casual
            </span>
          </p>
        </header>
        <section
          class="card-description"
        >
          <h2>
            Old Hickory Steakhouse
          </h2>
        </section>
        <section
          class="card-information"
        >
          <p
            class="location"
          >
            Oxon Hill, MD
          </p>
          <p
            class="phone"
          >
            (301) 965-4000
          </p>
        </section>
      </section>
      <section
        class="restaurant-card"
      >
        <header>
          <p
            class="genre"
          >
            Japanese
          </p>
          <p
            class="genre"
          >
            Sushi
          </p>
          <p
            class="genre"
          >
            Asian
          </p>
          <p
            class="genre"
          >
            Contemporary
          </p>
          <p
            class="genre"
          >
            Seafood
          </p>
          <p
            id="attire"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-briefcase fa-w-16 "
              data-icon="briefcase"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M320 336c0 8.84-7.16 16-16 16h-96c-8.84 0-16-7.16-16-16v-48H0v144c0 25.6 22.4 48 48 48h416c25.6 0 48-22.4 48-48V288H320v48zm144-208h-80V80c0-25.6-22.4-48-48-48H176c-25.6 0-48 22.4-48 48v48H48c-25.6 0-48 22.4-48 48v80h512v-80c0-25.6-22.4-48-48-48zm-144 0H192V96h128v32z"
                fill="currentColor"
              />
            </svg>
            <span
              class="label"
            >
              business casual
            </span>
          </p>
        </header>
        <section
          class="card-description"
        >
          <h2>
            Matsuhisa
          </h2>
        </section>
        <section
          class="card-information"
        >
          <p
            class="location"
          >
            Aspen, CO
          </p>
          <p
            class="phone"
          >
            (970) 544-6628
          </p>
        </section>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`RestaurantContainer should match the snapshot with the next pack of restaurants 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="restaurants-container"
      >
        <section
          class="no-card"
        >
          <img
            alt="cafe"
            src="cafe.svg"
          />
          <h2>
            No restaurant to display!
          </h2>
        </section>
        <footer
          class="pagination"
        >
          <button
            id="prev-page-btn"
          >
             previous page 
          </button>
          <p
            class="pages"
          >
            2 / 3
          </p>
          <button
            id="next-page-btn"
          >
             next page 
          </button>
        </footer>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="restaurants-container"
    >
      <section
        class="no-card"
      >
        <img
          alt="cafe"
          src="cafe.svg"
        />
        <h2>
          No restaurant to display!
        </h2>
      </section>
      <footer
        class="pagination"
      >
        <button
          id="prev-page-btn"
        >
           previous page 
        </button>
        <p
          class="pages"
        >
          2 / 3
        </p>
        <button
          id="next-page-btn"
        >
           next page 
        </button>
      </footer>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
